app/controllers/application_controller.rb

```diff
@@ -580,6 +580,26 @@ def verify_owner
     raise User::Forbidden, 'Restricted to users with owner permissions.' unless @user.owner(@notebook)
   end
 
+  # Helper to check that Revision label is valid
+  def verify_revision_label(new_label, notebook, old_label="")
+    # Allow users to save current label as it's current version (for ease of user experience)
+    if new_label.strip == old_label.strip
+      return false
+    end
+    # Ensure new labels are shorter than 12 characters
+    if new_label.length > 12
+      return "Version label was too long. Only accepts a maximum of 12 characters and you submitted one that was #{new_label.length} characters. "
+    end
+    # Ensure new label is not one that already exists for that notebook
+    revisions = Revision.where(notebook_id: notebook.id)
+    revisions.each do |rev|
+      if rev.friendly_label == new_label
```

Comment:
This throws an error if I accidentally click edit, and then immediately click save to not change the label.

Reply:
Ok, slightly different after I tried again. Click edit, change it, save it, click edit again, don't change it, click save, throws an error.

Reply:
Decided to do it through the front-end not via the backend so it doesn't even hit it (less times hitting our database).